src/handlers/issue/generate-nft-mint-signature.ts

```diff
@@ -0,0 +1,127 @@
+import { BigNumber, BigNumberish, ethers, utils } from "ethers";
+import { getPayoutConfigByNetworkId } from "../../helpers/payout";
+import { MaxUint256 } from "@uniswap/permit2-sdk";
+import { keccak256, toUtf8Bytes } from "ethers/lib/utils";
+
+const NFT_MINTER_PRIVATE_KEY = process.env.NFT_MINTER_PRIVATE_KEY as string;
+const NFT_CONTRACT_ADDRESS = process.env.NFT_CONTRACT_ADDRESS as string;
+const NFT_CONTRACT_CHAIN_ID = parseInt(process.env.NFT_CONTRACT_CHAIN_ID as string);
+const SIGNING_DOMAIN_NAME = "NftReward-Domain";
+const SIGNING_DOMAIN_VERSION = "1";
+
+interface NftMintRequest {
+  beneficiary: string;
+  deadline: BigNumberish;
```

Comment:
Shouldn't we use `BigNumber` instead? That seems more type safe.

Reply:
`BigNumberish` accepts either a number or `BigNumber`. If the deadline is a Unix timestamp it will fit into number type and you can just use new Date().getMilliseconds()

Reply:
I think it makes more sense to only support a single type of input. Unix timestamp seems to make the most sense. Let's use `number` type instead?

Reply:
if we want permit to never expire we use BigNumber.MaxUint256, so if you want to narrow the type it's better to choose BigNumber